scraper/tfs/__init__.py

```diff
@@ -0,0 +1,187 @@
+    return tfs_connection
+
+
+def create_tfs_project_analysis_client(url, token=None):
+    """
+    Create a project_analysis_client.py client for a Team Foundation Server Enterprise connection instance.
+    This is helpful for understanding project languages, but currently blank for all our test conditions.
+
+    If token is not provided, will attempt to use the TFS_API_TOKEN
+    environment variable if present.
+    """
+    if token is None:
+        token = os.environ.get('TFS_API_TOKEN', None)
+
+    tfs_connection = create_tfs_connection(url, token)
+    project_analysis_client = tfs_connection.get_client('vsts.project_analysis.v4_1.project_analysis_client.ProjectAnalysisClient')
+
+    if project_analysis_client is None:
+        msg = 'Unable to connect to TFS Enterprise (%s) with provided token.'
+        raise RuntimeError(msg, url)
+
+    return project_analysis_client
+
+
+def create_tfs_core_client(url, token=None):
+    """
+    Create a core_client.py client for a Team Foundation Server Enterprise connection instance
+
+    If token is not provided, will attempt to use the TFS_API_TOKEN
+    environment variable if present.
+    """
+    if token is None:
+        token = os.environ.get('TFS_API_TOKEN', None)
+
+    tfs_connection = create_tfs_connection(url, token)
+    tfs_client = tfs_connection.get_client('vsts.core.v4_1.core_client.CoreClient')
+
+    if tfs_client is None:
+        msg = 'Unable to connect to TFS Enterprise (%s) with provided token.'
+        raise RuntimeError(msg, url)
+
+    return tfs_client
+
+
+def create_tfs_git_client(url, token=None):
+    """
+    Creates a TFS Git Client to pull Git repo info
+    """
+    if token is None:
+        token = os.environ.get('TFS_API_TOKEN', None)
+
+    tfs_connection = create_tfs_connection(url, token)
+    tfs_git_client = tfs_connection.get_client('vsts.git.v4_1.git_client.GitClient')
+
+    if tfs_git_client is None:
+        msg = 'Unable to create TFS Git Client, failed to connect to TFS Enterprise (%s) with provided token.'
+        raise RuntimeError(msg, url)
+
+    return tfs_git_client
+
+
+def create_tfs_tfvc_client(url, token=None):
+    """
+    Creates a TFS TFVC Client to pull TFVC repo info
+    """
+    if token is None:
+        token = os.environ.get('TFS_API_TOKEN', None)
+
+    tfs_connection = create_tfs_connection(url, token)
+    tfs_tfvc_client = tfs_connection.get_client('vsts.tfvc.v4_1.tfvc_client.TfvcClient')
+
+    if tfs_tfvc_client is None:
+        msg = 'Unable to create TFS Git Client, failed to connect to TFS Enterprise (%s) with provided token.'
+        raise RuntimeError(msg, url)
+
+    return tfs_tfvc_client
+
+
+def get_all_projects(url, token):
+    """
+    Returns a list of all projects with their collection info from the server
+    """
+    project_list = []
+    tfs_client = create_tfs_core_client(url, token)
+
+    collections = tfs_client.get_project_collections(top=HARD_CODED_TOP)
```

Comment:
should the `top` be able to be passed in to the method? That way it's easier to override in the future (if necessary).

Maybe make it something like:

```python
def get_all_projects(url, token, top=HARD_CODED_TOP):
    # ... snip ...
    collections = tfs_client.get_project_collections(top=top)
```

You might also want to consider adding something to the docstring to talk about what this "top" is.

Reply:
Agree and done.